app/scripts/modules/ecs/src/serverGroup/configure/serverGroupConfiguration.service.ts

```diff
@@ -523,6 +524,16 @@ export class EcsServerGroupConfigurationService {
       }
     }
 
+    if (currentTargetGroups) {
+      const matched = intersection(allTargetGroups, currentTargetGroups);
+      const removedTargetGroups = xor(matched, currentTargetGroups);
+      if (removedTargetGroups && removedTargetGroups.length > 0) {
+        command.viewState.dirty.targetGroups = removedTargetGroups;
+      } else if(command.viewState.dirty && command.viewState.dirty.targetGroups) {
+        delete command.viewState.dirty.targetGroups;
```

Comment:
I'm not really sure what the impact of delete is here (are there potential negative repercussions to delete?), would it be better to set command.viewState.dirty.targetGroups = [] instead?

Reply:
So deleting the target group from the viewstate.dirty is the convention that we are following in rest of the code base. But setting it to [] also makes sense in order to just empty the dirty target groups. I will change it to set target groups to [] instead of deleting them all together.